Fix the site crash issue during the step of "hiding the thought" on the Start Tutorial

linghong opened the pull request:
Fix Ticket #1227

Cause: the incoming data is not always a string, thus the String.replace  function won't work correctly.

raineorshine reviewed (changes requested):
Thanks for identifying the cause. We want to take a slightly different approach on the solution. `stripTags` should only have to know about strings. It should not know anything about the `Child` type, i.e. `{ value: string, rank: number, ... }`. Notice the rest of the file and think about its purpose: all of the functions in the file operate on strings, and are not dependent on types like `Child` or `Context` that are specific to **em**'s data structure.

Also, polymorphism is *generally* a bad idea. We do use it in a couple places, e.g. `isRoot`, but we want to avoid it in favor of strict API's as much as possible.

Instead, identify the problem upstream that is passing a `Child` to `stripTags` instead of the expected string.

Notice that `stripTags` only accepts a string, yet is failing at runtime due to a bad type. This tells us that it must be being called by untyped code (i.e. `js` file). The tutorial contains some of the few `js` files left in the project. A good option here is to convert the caller to `ts` and let the compiler assist in showing you where the bad argument is being generated.

linghong commented:
New Changes:
1. The problem comes from two original functions on the TutorialStepAutoExpandExpand: The rootSubthoughtNotCursorWithSubthoughts and the rootGrandchildNotCursor
2. Removed multiple calls of those two functions, so that old smart phones could have better performance.
3. There are some spaces missing on the text, fixed it.

raineorshine commented on src/components/Tutorial/TutorialStepSuccess.jsx:
```diff
@@ -2,7 +2,7 @@ import React, { Fragment } from 'react'
 // eslint-disable-next-line jsdoc/require-jsdoc
 const TutorialStepSuccess = () => <Fragment>
-  <p>Huzzah... You have completed the tutorial. Creating thoughts and subthought is all you need to create a personalized thoughtspace. There's a lot more to <b>em</b> however. Additional commands have been designed to facilitate a truly fluid sensemaking experience.</p>
+  <p>Huzzah... You have completed the tutorial. Creating thoughts and subthought is all you need to create a personalized thoughtspace. There's a lot more to <b>em</b>; however, additional commands have been designed to facilitate a truly fluid sensemaking experience.</p>
```
The original punctuation was correct.

raineorshine commented on src/components/Tutorial/TutorialStepAutoExpandExpand.jsx:
```diff
@@ -1,29 +1,34 @@
+ * @param rootChildren The object array that show all the root thoughts.
+ * @returns The array that holds all the thoughts that that don't have a cursor, but have children.
+ */
+const ThoughtsNoCursorWithChild = (cursor, rootChildren) => {
```
Pascal case is only used on components and classes.

raineorshine reviewed (approved):
I completed the tutorial with no bugs. Thanks!
